packages/SwingSet/test/upgrade/test-upgrade.js

```diff
@@ -512,35 +473,23 @@ async function testMultiKindUpgradeChecks(t, mode, complaint) {
   t.teardown(c.shutdown);
   c.pinVatRoot('bootstrap');
   await c.run();
-
-  const run = async (method, args = []) => {
-    assert(Array.isArray(args));
-    const kpid = c.queueToVatRoot('bootstrap', method, args);
-    await c.run();
-    const status = c.kpStatus(kpid);
-    const result = c.kpResolution(kpid);
-    return [status, result];
-  };
+  const run = makeRun(c);
 
   // create initial version
-  const [v1status] = await run('buildV1WithMultiKind', [mode]);
-  t.is(v1status, 'fulfilled');
-
-  // upgrade should fail
-  if (complaint) {
-    console.log(`note: expect a '${complaint}' error below`);
-  }
-  const [v2status, v2result] = await run('upgradeV2Simple', [mode]);
-  if (complaint) {
-    t.is(v2status, 'rejected');
-    const e = kunser(v2result);
-    t.truthy(e instanceof Error);
-    t.regex(e.message, /vat-upgrade failure/);
-    // TODO: who should see the details of what v2 did wrong? calling
-    // vat? only the console?
-  } else {
-    t.is(v2status, 'fulfilled');
+  await run('buildV1WithMultiKind', [mode]);
+
+  // upgrade
+  const resultP = run('upgradeV2Simple', [mode]);
+  if (!complaint) {
+    await resultP;
+    t.pass();
+    return;
   }
+  console.log(`note: expect a '${complaint}' error below`);
```

Comment:
let's retain the `// TODO: who should see the details of what v2 did wrong? calling vat? only the console?` comment from the original, I think we still need to find an answer for that question